spec/lib/training/training_base_spec.rb

```diff
@@ -54,30 +53,6 @@
       end
     end
 
-    context 'when there are duplicate slugs' do
```

Comment:
These specs are testing important things. Maybe they can be moved to specs for the new class, instead of being tests of TrainingBase?